javascript-custom-methods/calculator.js

```diff
@@ -0,0 +1,34 @@
+/* exported calculator */
+var calculator = {
+  add: function (x, y) {
+    return x + y;
+  },
+  subtract: function (x, y) {
+    return x - y;
+  },
+  multiply: function (x, y) {
+    return x * y;
+  },
+  divide: function (x, y) {
+    return x / y;
+  },
+  square: function (x) {
+    return Math.pow(x, 2);
+  },
+  sumAll: function (numbers) {
+    var sum = 0;
+    for (var i = 0; i < numbers.length; i++) {
+      sum += numbers[i];
+    }
+    return sum;
+  },
+  getAverage: function (numbers) {
+    var average = 0;
+    var number = 0;
+    for (var i = 0; i < numbers.length; i++) {
+      number++;
+      average += numbers[i];
```

Comment:
This will work here, but you want to make sure to write this function so that it would work with any array. for example, if the array had the values `[2,4,2,6,5]` this solution would not work because `number` is only being incremented by one, and not by the value of the array.